doc/acl/Everflow-test-plan.md

```diff
@@ -0,0 +1,595 @@
+
+### Summary of the existing everflow test plan
+
+The existing everflow test plan only covers ingress ACL. And ingress everflow ACL table is binded to all ports. ACL rules are added to that ACL table. Then the PTF container is used for injecting packets. Two packets directions are tested: SPINE -> TOR ports and TOR -> SPINE ports. When the injected packets hit any of the configured ACL rules, the packets will be mirrored to configured mirror session. GRE tunnel is used for sending mirrored packets with src, dst IP and other parmeters configured in the mirror session. Below test cases are used for verifying that the DUT switch can properly forwared the mirrored packets according to different routing configurations for mirror session destination.
+
+* testcase 1 - Resolved route
+* testcase 2 - Longer prefix route with resolved next hop
+* testcase 3 - Remove longer prefix route
+* testcase 4 - Change neighbor MAC address
+* testcase 5 - Resolved ECMP route
+* testcase 6 - ECMP route change (add next hop)
+* testcase 7 - ECMP route change (remove next hop used by session)
+* testcase 8 - ECMP route change (remove next hop not used by session)
+
+### Extend the test plan to cover both ingress and egress ACL
+
+After egress ACL is supported, this test plan needs to cover both ingress and egress ACL. But for supporting swiches that do not support ingress yet, an ansible playbook variable is required for indicating egress ACL capability. Value for the variable can be passed from CLI.
+
+To cover egress ACL everflow testing, the ACL tables should not be binded to all ports. And different ACL tables for ingress and egress ACL need to be tested. Based on this requirement, totally 4 ACL tables should be configured before testing:
+
+ACL Table | Type | Bind to | Stage | Packet flow | Description
+----------|------|---------|-------|-------------|------------
+TORIN | MIRROR | TOR ports | Ingress | TOR -> SPINE | Ingress ACL, packets injected into TOR ports, hit ACL, mirrored to mirror session destination
+SPINEIN | MIRROR | SPINE ports | Ingress | SPINE -> TOR | Ingress ACL, packets injected into SPINE ports, hit ACL, mirrored to mirror session destination
+TOROUT | MIRROR | TOR ports | Egress | SPINE -> TOR | Egress ACL, packets injected into SPINE ports, forwarded to TOR ports, hit ACL, mirrored to mirror session destination
+SPINEOUT | MIRROR | SPINE ports | Egress | TOR -> SPINE | Egress ACL, packets injected into TOR ports, forwarded to SPINE ports, hit ACL, mirrored to mirror session destination
+
+For switches do not support egress ACL, only the first two ACL tables will be configured for testing.
+
+### Related **DUT** CLI commands
+
+The existing test plan uses a script calling swsssdk to configure mirror session. Calling SDK to configure DUT switch is not user friendly. Now DUT switch supports configuring mirror session using CLI command `config mirror_session`.
```

Comment:
Agree. Could you please help remove the current mirror_session.py script out of the code.

Reply:
Sure, I will.